server/sonar-server/src/main/java/org/sonar/server/batch/ProjectAction.java

```diff
@@ -59,20 +66,97 @@ public void define(WebService.NewController controller) {
       .setExampleValue("SonarQube Way");
 
     action
-      .createParam(PARAM_PREVIEW)
-      .setDescription("Preview mode or not")
+      .createParam(PARAM_ISSUES_MODE)
+      .setDescription("Issues mode or not")
       .setDefaultValue(false)
       .setBooleanPossibleValues();
   }
 
   @Override
-  public void handle(Request request, Response response) throws Exception {
-    ProjectRepositories ref = projectReferentialsLoader.load(ProjectRepositoryQuery.create()
-      .setModuleKey(request.mandatoryParam(PARAM_KEY))
-      .setProfileName(request.param(PARAM_PROFILE))
-      .setPreview(request.mandatoryParamAsBoolean(PARAM_PREVIEW)));
-    response.stream().setMediaType(MimeTypes.JSON);
-    IOUtils.write(ref.toJson(), response.stream().output());
+  public void handle(Request wsRequest, Response wsResponse) throws Exception {
+    ProjectRepositories data = projectDataLoader.load(ProjectDataQuery.create()
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) org.sonar.server.batch.ProjectAction must not use org.sonar.batch.protocol.input.ProjectRepositories [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AThe_Server_stack_should_not_access_classes_of_the_Batch_stack)